### ISSUE TITLE
Comments don't work on class diagrams

### ISSUE THREAD

**Issue by Dunning-Kruger** [mermaid-js/mermaid#1073]
[This](https://mermaidjs.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoiY2xhc3NEaWFncmFtXG4lJSBhc2RcbmNsYXNzIEJhciB7XG59XG4iLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCJ9fQ/error/UGFyc2UgZXJyb3Igb24gbGluZSAyOgpjbGFzc0RpYWdyYW0lJSBhc2RjbGFzcyBCYXIge30KLS0tLS0tLS0tLS0tLS0tLS0tXgpFeHBlY3RpbmcgJ0NMQVNTJywgJ0FOTk9UQVRJT05fU1RBUlQnLCAnTUVNQkVSJywgJ1NFUEFSQVRPUicsICdVTklDT0RFX1RFWFQnLCAnTlVNJywgJ0FMUEhBJywgZ290ICdORVdMSU5FJw) does not work.
[This ](https://mermaidjs.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoiZ3JhcGggVERcbkFbQ2hyaXN0bWFzXSAtLT58R2V0IG1vbmV5fCBCKEdvIHNob3BwaW5nKVxuJSUgYXNkYXNkXG5CIC0tPiBDe0xldCBtZSB0aGlua31cbiUlIGFzZGFzZFxuQyAtLT58T25lfCBEW0xhcHRvcF1cbkMgLS0-fFR3b3wgRVtpUGhvbmVdXG5DIC0tPnxUaHJlZXwgRltmYTpmYS1jYXIgQ2FyXVxuIiwibWVybWFpZCI6eyJ0aGVtZSI6ImRlZmF1bHQifX0) does work.

Original issue with the implementation of comments: [here](https://github.com/mermaid-js/mermaid/issues/47).

**Comment by jgreywolf** [comment]
Interestingly - if you remove the `%%` before the comment, it does work (meaning, the diagram renders, and the text does not display)

[See here](https://mermaidjs.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoiY2xhc3NEaWFncmFtXG5hc2RcbmNsYXNzIEJhciB7XG5pbnQgbWVtYmVyXG59XG4iLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCJ9fQ)

**Comment by lukewlms** [comment]
@jgreywolf Interesting workaround!  I'm not confident that will continue to work long-term though.  For now we're checking comments into code at the top, but not pasting them in when we render.  Would be convenient to be able to intersperse comments where appropriate as spec'd.

**Comment by Dunning-Kruger** [comment]
I think, assuming it's feasible, all graphs should share the comment syntax.

**Comment by jgreywolf** [comment]
Im working on this one :)

**Comment by jgreywolf** [comment]
I should say, I have started looking at this one.  Still trying to wrap my head around the logic flow within Jison, and how things are actually being parsed.  Porting functionality from flowcharts, but doesnt appear to be an easy mapping :)

**Comment by jgreywolf** [comment]
Well, Im more than a little confused now.  I've been trying to port the way that comments are handled in flowcharts (based on jison), and it does not seem to be working - meaning I get something like the following:
```
Parse error on line 2:
    ...sDiagram%% Comment class Class1 {int 
    ----------------------^
    Expecting 'CLASS', 'ANNOTATION_START', 'MEMBER', 'SEPARATOR', 'UNICODE_TEXT', 'NUM', 'ALPHA', got 'NEWLINE'
```
 As I started reading more about bison/jison I noticed that how the `flow.jison` file is setup did not seem to match what I would have expected.  So I started removing the comment related code in that file...   and even without that code the tests were still working fone (meaning, they were parsing comments)

Well, that is even more odd to me.  Just weird.  Ive done my best to replicate code between the two, but still not sure what I am missing.  

If anyone is interested in taking a look at where I am, see my [fork here](https://github.com/jgreywolf/mermaid/tree/1073-CommentsInClassDiagrams)
